### PR TITLE
Create Thinking in Redux category and move some pages in it

### PR DESCRIPTION
**Pull request by sbaudray**
---
name: :book: Thinking in Redux
about: Adding a category and moving some pages to it
---

## PR Type

**Does this PR add a _new_ page, or update an _existing_ page?**

It moves some pages to a new category

## Checklist

- [x] Is there an existing issue for this PR?
  - https://github.com/reduxjs/redux/issues/3594
- [x] Have the files been linted and formatted? 

## What docs page is being added or updated?

- **Section**: Added Thinking in Redux
- **Page**: 
    - introduction/motivation -> thinking-in-redux/motivation
    - introduction/three-principles -> thinking-in-redux/three-principles
    - introduction-prior-art -> thinking-in-redux/prior-art

### CODE REVIEW

File: website/_redirects [hunk 1]
```diff
@@ -65,11 +65,11 @@
 
 # Old old Gitbook links?
 /docs/introduction/CoreConcepts.html        /introduction/core-concepts
```
Comment (markerikson) [inline website/_redirects:67]: :exclamation: These are the very old legacy Gitbook links.  We do need to update these, but we _also_ need to add new redirects from the _current_ URLs as well.

Can you start a new comment-block-ish section somewhere labeled "2020 site reorganization" or soemthing along those lines, and add new redirect entries there from `/introduction/` to `/thinking-in-redux/` for these pages?

**Comment by markerikson** [comment]
Hey, thanks for filing this!

Got a couple items I'd like to discuss around intended organization before this is ready.

First, the other relevant PR here is #3600 , which talks about adding an "Understanding Redux" section that would have categories inside it like "Thinking in Redux" and "History and Design".

Second, given that, I'd like to see this new "Thinking in Redux" category as a sub-category.

Third, I'm not yet 100% sure I want to name the parent category "Understanding Redux", but I'm not immediately sure what a better option is.  Suggestions?

**Comment by markerikson** [comment]
Having thought about it further, I'm leaning towards:

- New parent category, "Understanding Redux", at `/understanding/`
  - New child category, "Thinking in Redux", at `/understanding/thinking-in-redux/`
    - Motivation
    - Three Principles
  - New child category, "History and Design"
    - Prior Art

File: website/_redirects (hunk 1 above)
Comment (sbaudray) [reply website/_redirects:67]: It should be better now

**Comment by markerikson** [comment]
Awright, this looks pretty good.  Confirmed the redirect for the "Motivation" page works correctly. Let's get this in.

**Comment by markerikson** [comment]
Hah. And of course, as soon as I hit the "Merge" button, I remembered that I wasn't 100% sure on having the "Understanding Redux" category be right below the "Introduction".

@sbaudray : Tell you what.  Can you create a second PR that:

- Moves the "Understanding" category to be after "Recipes" (ie, 4th in the sidebar list)
- Moves the "Glossary" page from the "Other" category to "Understanding > Thinking in Redux"
- Moves the "Troubleshooting" page from the "Other" category to "Recipes"
- Removes the "Other" category entirely

From there, we can think about how to reorganize the "Introduction" and "Recipes" categories further.

**Pull request merged**